mm2src/coins/eth.rs

```diff
@@ -535,56 +539,34 @@ pub type Web3RpcFut<T> = Box<dyn Future<Item = T, Error = MmError<Web3RpcError>>
 pub type Web3RpcResult<T> = Result<T, MmError<Web3RpcError>>;
 type EthPrivKeyPolicy = PrivKeyPolicy<KeyPair>;
 
-#[derive(Clone, Debug)]
-pub(crate) struct LegacyGasPrice {
-    pub(crate) gas_price: U256,
-}
-
-#[derive(Clone, Debug)]
-pub(crate) struct Eip1559FeePerGas {
-    pub(crate) max_fee_per_gas: U256,
-    pub(crate) max_priority_fee_per_gas: U256,
-}
-
 /// Internal structure describing how transaction pays for gas unit:
 /// either legacy gas price or EIP-1559 fee per gas
 #[derive(Clone, Debug)]
 pub(crate) enum PayForGasOption {
-    Legacy(LegacyGasPrice),
-    Eip1559(Eip1559FeePerGas),
+    Legacy {
+        gas_price: U256,
+    },
+    Eip1559 {
```

Comment:
As I mentioned [here](https://github.com/KomodoPlatform/komodo-defi-framework/pull/2533#discussion_r2221577248) the term `Legacy` is not very clear I think, especially for those unfamiliar with eth internals. At first glance I actually thought it referred to our own codebase as my ETH knowledge is limited.

What do you think about renaming it to `Type0` instead?

That would make it look like this:

```rust
    /// Also known as "legacy".
    /// Ref: https://ethereum.org/en/developers/docs/transactions/#typed-transaction-envelope
    Type0 {
        gas_price: U256,
    },
```

With that, it shouldn't be possible to get confused even if you are not familiar with the internals of the network.

Reply:
Here we refer to gas price, not the transaction type.
Priority transactions were introduced in the 'Type2' transaction, so there is also Type1.
Plus there is no 'type0' actually: no 'types' existed initially at all for transactions so they call them 'legacy' in docs.
I.e., 'legacy transaction’s gas_price' is a term used in the EIP-1559 doc.

Reply:
Can you link a doc so I can read more about legacy gas price?